public/app/db.js

```diff
@@ -58,14 +58,11 @@ const bootstrapDatabase = (dbPath = DATABASE_PATH) => {
   return db;
 };
 
-const ensureAlgorithmsExist = async (
-  db,
-  algorithmsFolder = ALGORITHMS_FOLDER,
-) => {
+const ensureAlgorithmsExist = async (db) => {
   try {
     // create the algorithms folder if it doesn't already exist
-    if (!fs.existsSync(algorithmsFolder)) {
-      await mkdirp(algorithmsFolder);
+    if (!fs.existsSync(ALGORITHMS_FOLDER)) {
```

Comment:
I think you can  refactor it with `fs-extra` and `ensureDirSync` or something similar. There is an example in `test/application.js`